src/auxiliary/mpi_arrays.jl

```diff
@@ -0,0 +1,332 @@
+
+# TODO: MPI. Keep this module inside Trixi or move it to another repo as
+#            external dependency with simple test suite and documentation?
+module TrixiMPIArrays
+
+using ArrayInterface: ArrayInterface
+using MPI: MPI
+
+import ..Trixi: mpi_comm, mpi_rank
```

Comment:
Why not `using`?

Reply:
I'm extending them, so I would need something like
```julia
using ..Trixi

...

Trixi.mpi_comm(...) = ...
```
Any preference?

Reply:
No preference, no. I was just asking out of curiosity, but this explains it. Personally, I like the `Trixi.mpi_comm(...)` syntax better since the reader explicitly knows that this extends an existing functionality, but I don't think it matters much.